### ISSUE TITLE
default queues-poll-interval seems to be 4s instead of 1s

### ISSUE THREAD

**Issue by pekeler** [arangodb/arangodb#1959]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x] 2.8
- [ ] 3.0
- [ ] self-compiled devel branch

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [ ] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [x] MacOS, version: 10.11.5

### Foxx

According to the docs, the default queues-poll-interval is 1 second. With that setting, I'd expect the average time between me putting a job into the queue and the job starting to be processed to be just over half a second. But it is consistently exactly 2 seconds.

mycontroller.js:

```JavaScript
let myQueue = Foxx.queues.create('myjob');

controller.put('/myaction', function(req, res, injected) {
  myQueue.push({mount: '/myjob', name: 'myjob'}, {what: 'ever'});
  console.log('enqueued...');
```

myjob.js:

```JavaScript
  console.log('processing...');
```

The timestamps between "enqueued" and "processing" in my log are always 2s apart.

```
2016-07-22T19:03:44Z [47953] INFO /myapp, incoming request from 127.0.0.1: PUT http://localhost/_db/mydb/myapp/mycontroller/myaction
2016-07-22T19:03:44Z [47953] INFO enqueued...
2016-07-22T19:03:44Z [47953] INFO /myapp, outgoing response with status 200 of type undefined, no body
2016-07-22T19:03:46Z [47953] INFO processing...
```

The 2.8 docs state
> If Foxx queues are not used much, then this value may be increased to make the queues thread wake up less.

But even if I repeat the above test every 2s for ten times, it's still always exactly 2 seconds between enqueuing and processing. Why?

**Comment by pekeler** [comment]
Starting it up with `arangod --server.foxx-queues-poll-interval 0.1` I can bring it down to a 1s delay most of the time (2 out of 20 attempts I still get a 2s delay). At a cost of 50% cpu load :/

Truncating the _jobs collection doesn't seem to make a difference.

**Comment by jsteemann** [comment]
Looks like the queue manager is woken up every `--server.foxx-queues-poll-interval` seconds. If set to `1` the queue manager actually wakes up every second. However, when it finds a job, it seems to schedule it with a delay of 1 extra second (for whatever reason). The following patch should correct that, so the dispatched job will be scheduled for immediate execution:
```diff
diff --git a/js/server/modules/org/arangodb/foxx/queues/manager.js b/js/server/modules/org/arangodb/foxx/queues/manager.js
index 6383270..d371d34 100644
--- a/js/server/modules/org/arangodb/foxx/queues/manager.js
+++ b/js/server/modules/org/arangodb/foxx/queues/manager.js
@@ -87,7 +87,7 @@ var runInDatabase = function () {
               } catch(e) {}
               db._useDatabase(initialDatabase);
             },
-            offset: 1,
+            offset: 0,
             isSystem: true,
             params: {
               job: _.extend({}, job, {status: 'progress'}),
```

**Comment by jsteemann** [comment]
Extra delay removed in 2.8, 3.0 and devel branches.

**Comment by jsteemann** [comment]
The extra delay of 1 second issue is fixed in 3.0.4, which was released today.
There is also a fix for 2.8 via c7a7b1141a13f3593a26f20bedda3e2f03555f43, but there is no release date yet.

**Comment by pluma** [comment]
Closing this as fixed in 3.0 and 2.8.
